crates/oxc_linter/src/rules/eslint/no_this_before_super.rs

```diff
@@ -246,6 +251,27 @@ impl NoThisBeforeSuper {
             }),
         })
     }
+
+    fn contains_this_or_super(arg: &Argument, semantic: &Rc<Semantic>) -> bool {
```

Comment:
You are not using this `semantic` parameter, I think you can safely remove it.

Reply:
The method `Self::contains_this_or_super_in_args(&call_expr.arguments, semantic)` is used to recursively handle cases like `super(a(b(this.c)))`.

Reply:
You are absolutely right, Semantic is needed to be passed around, however, that function only uses its `semantic` to call `Self::contains_this_or_super`:

```rust
    fn contains_this_or_super_in_args(args: &[Argument], semantic: &Rc<Semantic>) -> bool {
        args.iter().any(|arg| Self::contains_this_or_super(arg, semantic))
    }
```

And `Self::contains_this_or_super` only uses it to call `Self::contains_this_or_super_in_args`.
I think you've gotten rid of the need for semantic usage however the compiler because of cyclic recursion doesn't notice this to provide a dead_code warning.

Please let me know if I'm missing something.